cmd/object-handlers.go

```diff
@@ -1113,21 +1110,21 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 		pipeReader, pipeWriter := io.Pipe()
 		snappyWriter := snappy.NewWriter(pipeWriter)
 
-		actualReader, err := hash.NewReader(reader, size, md5hex, sha256hex, actualSize)
+		var actualReader *hash.Reader
+		actualReader, err = hash.NewReader(reader, size, md5hex, sha256hex, actualSize)
 		if err != nil {
 			writeErrorResponse(w, toAPIErrorCode(err), r.URL)
 			return
 		}
+
 		go func() {
-			defer pipeWriter.Close()
-			defer snappyWriter.Close()
 			// Writing to the compressed writer.
-			_, err = io.CopyN(snappyWriter, actualReader, actualSize)
-			if err != nil {
-				hashError = err
-				return
-			}
+			_, cerr := io.CopyN(snappyWriter, actualReader, actualSize)
```

Comment:
shouldn't we treat `io.ErrUnexpectedEOF` seperately as ErrIncompleteBody ? as per this conversation [https://github.com/minio/minio/pull/6079#discussion_r206027174](url)

Reply:
No, we shouldn't PutObject is perfectly capable of returning this error. it is the pipeReader which sees this error.

Reply:
ok